uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/AbsenceType.php

```diff
@@ -275,6 +271,25 @@ private function getDAOFieldAttributes($field) {
   }
 
   /**
+   * Returns parameters needed for the calculation unit selector
+   *
+   * @return array
+   */
+  private function getCalculationUnitSelectorParams () {
+    $calculationUnitSelectorParams = [
+      'options' => $this->getCalculationUnitOptions(),
+      'option_url' => NULL,
+      'label' => ts('Calculate Leave in')
+    ];
```

Comment:
The function `getCalculationUnitOptions()` is no longer needed, the reason I created the function was to find a workaround to not disable the calculation unit field and make sure that the selected option is the only option.

We can simply change the the `$calculationUnitSelectorParams` array to

> $calculationUnitSelectorParams= [
>       'option_url' => NULL,
 >      'label' => ts('Calculate Leave in')
 >    ];

Civi will automatically build the select options from the calculation unit option group.
Also we should remove the `getCalculationUnitOptions()` function.

Reply:
Done.